### ISSUE TITLE
Is it possible to overfit?

### ISSUE THREAD

**Issue by see--** [qqwweee/keras-yolo3#33]
Thanks for this great repo! I wanted to use `train.py` to overfit to a small sample. In fact the loss goes down to ~25 but the predicted boxes are bad. Did you manage to overfit/train with good results? Here is what I am using:
### train.txt
```
img_000.jpg 119,349,787,844,0 811,249,1476,846,0
img_001.jpg 7,18,64,76,0 88,24,147,77,0 144,37,190,84,0 163,24,217,78,0 ,151,146,0 128,137,182,190,0 162,107,208,156,0 184,149,214,178,0 147,188,200,225,0 8,139,221,0 38,156,106,214,0 22,96,91,157,0 75,76,151,149,0
img_002.jpg 23,37,199,201,0
img_003.jpg 14,194,273,448,1 162,80,396,330,1 269,203,525,459,1
img_004.jpg 19,79,116,169,1 76,52,154,147,1 118,74,206,169,1
```

### labels.txt
```
tomato
apple
```
![img_000](https://user-images.githubusercontent.com/11165293/39665098-dd9c5a44-508e-11e8-8e7d-45566e1c235a.jpg)
![img_001](https://user-images.githubusercontent.com/11165293/39665099-ddc06d12-508e-11e8-963b-6961eb3055a7.jpg)
![img_002](https://user-images.githubusercontent.com/11165293/39665100-ddda3152-508e-11e8-98e6-e8adbc37d5b5.jpg)
![img_003](https://user-images.githubusercontent.com/11165293/39665101-ddf95f28-508e-11e8-8f50-18828c0fcbd5.jpg)
![img_004](https://user-images.githubusercontent.com/11165293/39665102-de199ee6-508e-11e8-946a-6fdce62f7e0d.jpg)

```

**Comment by see--** [comment]
I am closing this as it works with pascal VOC. You can just train longer. For example on VOC I have: `loss: 12.3151 - val_loss: 13.3674` after 30 epochs.

**Comment by qqwweee** [comment]
I've tried your dataset, 2000 epochs with val=0,   and batch_size=1 for the last 1000 epochs.
I've found that when `K.learning_phase()=1` in `yolo.py  sess.run`, the predict is perfect, but 0 is bad.

That is because of the different behavior of BatchNormalization layers in `train` or `test` mode.
We freeze the BN weights, so it cannot learn to fit new data distribution. I believe the original BN weights are good for pictures in common life, but not good for this little dataset. BN actually increases generalization of model, and harms overfitting in this case.
